File: NativeAPI/Python/b2_set_bucket_notification_rules.py
import requests
import json

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
bucket_id = ""  # Provided by b2_list_buckets
event_types = [ "b2:ObjectCreated:Upload", "b2:ObjectCreated:MultipartUpload" ]  # Events that will trigger the rule
rule_name = ""  # The name of the rule you want to create
prefix = ""  # The file name prefix to which the rule applies
url = "https://www.example.com"  # Your webhook URL

response = requests.post(
    f"{api_url}/b2api/v4/b2_set_bucket_notification_rules",
    headers={"Authorization": account_authorization_token},
    json={
        "bucketId": bucket_id,
        "eventNotificationRules": [
            {
                "eventTypes": event_types,
                "isEnabled": True,
                "name": rule_name,
                "objectNamePrefix": prefix,
                "targetConfiguration": {
                    "targetType": "webhook",
                    "url": url
                }
            }
        ]
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2SetBucketNotificationRules.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2SetBucketNotificationRules {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String bucketId = ""; // Provided by b2_create_bucket or b2_list_buckets
        String eventTypes = "[" + // Event types that will trigger the rule
                "\"b2:ObjectCreated:Upload\"," +
                "\"b2:ObjectCreated:MultipartUpload\"" +
                "]";
        String ruleName = ""; // Name of the rule you are creating
        String prefix = ""; // Restrict rule to files with names starting with this prefix (optional)
        String url = ""; // Your webhook URL

        String postParams = "{" +
                "\"bucketId\": \"" + bucketId + "\"," +
                "\"eventNotificationRules\": [" +
                "{" +
                "\"eventTypes\": \"" + eventTypes + "\"," +
                "\"isEnabled\": true," +
                "\"name\": \"" + ruleName + "\"," +
                "\"objectNamePrefix\": \"" + prefix + "\"," +
                "\"targetConfiguration\": {" +
                "\"targetType\": \"webhook\"," +
                "\"url\": \"" + url + "\"" +
                "}" +
                "}" +
                "]" +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_set_bucket_notification_rules"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
